### DISCUSSION TITLE
Utils for API endpoints

### DISCUSSION THREAD

**Discussion by bastienrobert** [nuxt/nuxt#18343] [Ideas]
Is there any plan for adding a default middleware for body/params/querystring parsing on the `req` object and status setting, redirect, json sending... for the `res` object on `/server/api` files?

NextJS has an implementation for this, which is really useful during API development: [response helpers doc](https://nextjs.org/docs/api-routes/response-helpers), [api middlewares doc](https://nextjs.org/docs/api-routes/api-middlewares), 
[next/server/api-utils.ts](https://github.com/vercel/next.js/blob/canary/packages/next/server/api-utils.ts).

**Comment by pi0** [comment]
Hi @bastienrobert. Yes! You can use [utils from h3](https://github.com/unjs/h3#utilities) directly in middleware :)
